### PR TITLE
Landing Page Improvement - "How Mappypals works" Section

### PR DESCRIPTION
**Pull request by danielgnz**
Trello task: https://trello.com/c/7DzTWRrH/183-landing-page-improvement

- Translated landing page mockup to code
- Added reusable components: Grid, Text and Container
- Created responsive design (mobile first)

Should you have any concerns, or comments, please let me know. Also, if there's anything that needs improvement, I'd be happy to do it.

### CODE REVIEW

File: src/components/HomeCard/home-card.component.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+
+import Text from '../../utils/components/Text';
```
Comment (ArnasDickus) [inline src/components/HomeCard/home-card.component.jsx]: All files are named with js extension. Why did you add a jsx extension instead? 
https://reactjs.org/docs/react-without-jsx.html

File: src/components/HomeCard/home-card.component.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                        { text }
+                    </Text>
+                    
+                </Column>
+            </Row>
+        </Card>
+    )
+}
+
+export default HomeCard;
```
Comment (ArnasDickus) [inline src/components/HomeCard/home-card.component.jsx]: This ending needs a new line. Try running `npm run format`

File: src/components/HomeCard/home-card.styles.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import styled from 'styled-components';
+import Container from '../../utils/components/Container';
+
+
+export const Circle = styled.div`
```
Comment (ArnasDickus) [inline src/components/HomeCard/home-card.styles.jsx]: All files are named with js extension. Why did you add a jsx extension instead?
https://reactjs.org/docs/react-without-jsx.html

File: src/components/HomeCard/home-card.component.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    Circle,
+    Card
+} from './home-card.styles';
+
+export const HomeCard = ({ number, title, icon, text, ...otherProps}) => {
+
+    const fontSize = otherProps.textSize ? otherProps.textSize : 1;
+
+    return (
+        <Card>
```
Comment (ArnasDickus) [inline src/components/HomeCard/home-card.component.jsx]: I see you made it reusable. Good job. I didn't think of doing that.

File: src/components/HomeCard/home-card.styles.jsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    height: 30px;
+    width: 30px;
+    color: #fff;
+    background-color: #6831DE;
+    border-radius: 100%;
+    font-weight: bold;
+    font-family: 'Poppins', sans-serif;
```
Comment (ArnasDickus) [inline src/components/HomeCard/home-card.styles.jsx]: A little confused here, you hardcoded font-family, however in-home card components, you also add additional, Poppins family property. I don't think you need it here.

File: src/pages/Home/Home.js [hunk 6]
```diff
@@ -1,39 +1,178 @@
...
 
+import useWindowSize from '../../utils/effects/useWindowSize.effect';
+
 const Home = user => {
-    return true ? (
+    const size = useWindowSize();
+
+    const [fontSize, setFontSize] = useState(1);
+
+    useEffect(() => {
```
Comment (ArnasDickus) [inline src/pages/Home/Home.js:35]: So this is the javascript way to write media queries? Looks complicated. I don't think this is the wrong way, however, I think it would be easier just to use regular media queries.  However, I'm not going to suggest it for you to change it. Navbar was created a similiar way.

File: src/pages/Home/Home.style.jsx [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+`;
+
+export const StyledContainer = styled(Container)`
+    background: #fff;
+    @media only screen and (min-width: 1200px) {
+        background: url(${bgImage});
+        background-repeat: no-repeat;
+        background-position: center;
+    }
+`;
```
Comment (ArnasDickus) [inline src/pages/Home/Home.style.jsx]: Need formating. Try running `npm run format`

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
Hello, thanks for adding PR.

It seems there was a misunderstanding. Landing page how mappypals work should replace the old landing page. Not be an additional section.
1) Remove old landing page.
2) Adjust if any, for new landing page to look similiar to mockup.

The utils folder was created by phpist, for things like local storage, and redux. You added in there components and effects folder. I think your components and effects file should be better put in components>UI>specific folder. What do you think?

Comparing your design to Trello, I would decrease its height, you should see sign in button without the need to scroll.

As I mentioned in the comments below. Format your file, and change jsx extensions to js.

File: src/components/HomeCard/home-card.styles.jsx (hunk 3 above)
Comment (danielgnz) [reply src/components/HomeCard/home-card.styles.jsx]: I added the .jsx extension because I used two extensions. (style and jsx)

When you create a file _home-card.component_ or _home-card.styles_ and then you import them using ``import HomeCard from "./home-card.component"`` by default the compiler will be looking for .jsx extensions instead of .js and will throw an error if you use ``home-card.component.js``

However, if you want to use the .js extension, a quick fix would be changing the import statement to ``import HomeCard from "./home-card.component.js"``

This is just a personal preference. I like separating my code into .component and .styles so I know exactly what each file is for.

However, I changed the file extensions to .js, as you requested.

File: src/components/HomeCard/home-card.styles.jsx (hunk 5 above)
Comment (danielgnz) [reply src/components/HomeCard/home-card.styles.jsx]: Yes, I hardcoded it because I wanted the number inside each Circle to have the Poppins font-family. If I removed it, the number would have a default font-family of ``sans-serif``

In ``home-card.component.jsx`` I added the Poppins font family prop to the Text component, which is the description below the Icon.

File: src/pages/Home/Home.js (hunk 6 above)
Comment (danielgnz) [reply src/pages/Home/Home.js:35]: Yes, that's correct. This is the media query implementation using React hooks.

I used the hooks because I wanted to make the font size adjust instantly whenever the screen size changes.

In this scenario, I consider this approach to be better than media queries because:
1. It separates the concerns, i.e: if you wanted to add something in the future or change the font sizes, it's very easy to do so. You don't need to look through a big CSS file
2. I only used it for font-sizes so I thought using Hooks would make more sense.
3. Reusable. You can use this hook in other components.

PS: To be honest, I've been doing a course about React lately, and I wanted to practice using Hooks more.

You can use media queries if you want.

**Comment by danielgnz** [comment]
Hi, Arnas. Can you take a look at my latest commit and let me know if there’s anything that needs to be updated? Thank you

File: src/pages/Home/Home.js (hunk 6 above)
Comment (ArnasDickus) [reply src/pages/Home/Home.js:35]: I would argue, that developers who don't know react hooks, this would look confusing compared to media queries. The css files are not really that big. Media queries can also be reused. However, there is no point in arguing. Your way is also valid.

**Review by ArnasDickus — CHANGES_REQUESTED** [review]
So everything works fine. Except the landing page is too tall. Doesn't look like the [mockup](https://trello.com/c/2go1h860/152-landing-page-improvement). 
landing-page-height

**Comment by InteriorNature** [comment]
On my machine, it does the same. In Chrome it shows that at the root level grid columns are happening before the viewport is specified.

**Comment by danielgnz** [comment]
I'm fixing this issue now.

**Comment by ArnasDickus** [comment]
Hello, @danielgnz Travis test didn't pass. You need to fix that.
Line 8:  'bgImage' is defined but never used

**Review by ArnasDickus — APPROVED** [review]

**Comment by ArnasDickus** [comment]
Congratulations, I'm out of concerns.
